code/modules/mining/machine_vending.dm

```diff
@@ -0,0 +1,218 @@
+//PROGRESSION SYSTEM: Miners start with a basic ass manual pickaxe. They can shortly buy an autodrill, or save up for a /TG/ Kinetic Accelerator.
+//HARDSUITS: Miners all start with an *empty* industrial hardsuit. Many of the items they can buy here (auto-drill, kinetic accelerator) can be converted into modules
+//			with a special machine.
+//VENDOR: Obj/item's are immediately dispensed by the vendor. Machines, structures, vehicles, and etc get spawned on the cargo shuttle.
+//Think up of lots of items. Not everything needs to be unique or even mining-special, but it should be neat. Convert most of /tg/'s items. 25% of this at least
+//	should be bling. Things that shorten the distance between base and mining. Instant-teleporters should be one use.
+
+
+/obj/machinery/mineral/equipment_vendor
+	name = "mining equipment vendor"
+	desc = "An equipment vendor for miners, points collected at an ore redemption machine can be spent here."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "mining"
+	density = 1
+	anchored = 1
+	var/obj/item/weapon/card/id/inserted_id
+	var/list/prize_list = list( //keep in order of price
```

Comment:
I planned at one point to have research be able to upgrade the machines for even higher tiers items, but I never created enough content. Could be global at this point, yea.